### ISSUE TITLE
unable to create .binaryproto for vision kit

### ISSUE THREAD

**Issue by jass-singh-zz** [google/aiyprojects-raspbian#484]
I wanted to convert pretrained model (mobilenet_v1_0.5_160_quant_v1.pb) into .binaryproto,
commands used :- 

./bonnet_model_compiler.par \
  --frozen_graph_path=./mobilenet_v1_0.5_160_quant_frozen.pb \
  --output_graph_path=./mobilenet_v1_0.5_160_quant_v1.binaryproto \
  --input_tensor_name="Placeholder" \
  --output_tensor_names="final_result" \
  --input_tensor_size=160 \
  --debug

It shows error :- 
F graph_conversion/prune.cc:63] Output node named "final_result" does not appear in the graph.

If i vsiualize graph on  tensorboard, 
![graph_large_attrs_key _too_large_attrs limit_attr_size 1024 run](https://user-images.githubusercontent.com/20219876/45204358-963ca800-b29c-11e8-838b-a478fbb67271.png)

we can see last node final_result .
So, what should i do to resolve this, 
Thanks

**Comment by weiran-work** [comment]
@jass-singh 
First of all, VisionKit does not support quantized model.

We did release some pretrained models, you can download this file
https://dl.google.com/dl/aiyprojects/vision/tf_models_latest.tgz

Decompress it and you can find `README.md`.  and use the corresponding input tensor and output tensors name in the table.

For your graph, pass
--input_tensor_name="input" 
--output_tensor_names="MobilenetV1/Predictions/Softmax" 

for mobilenet_v1_160res_0.5_imagenet.pb

**Comment by jass-singh-zz** [comment]
Thanks @weiranzhao ,
using those commands i was able to convert successfully,

Actually i want to initially detect the animal like dog using 1 model , than perform classification on it using classifier by passing only that part of image which would be detected as dog by the model. At last display label with the breed of dog.

So, is it possible to combine 2 models in Vision kit ?
If yes , please suggest some method.

Thanks,

**Comment by weiran-work** [comment]
@jass-singh 

There are 2 methods I can think of to achieve what you need.

1) retrain a detector, that can recognize more dog breeds.
2) Have 2 separate models, a) detector b) classifier. You need to run detector on the input stream, and get result (bounding box and image) from VisionBonnet (the accelerator on top of pi board). Then you can crop the image based on bounding box  on pi and then send the image to accelerator to analyze with classifier and get result.

I'd suggest method 1) because it will give you better FPS. You can find retraining related instructions here.
search `How to train and deploy a customized object detection model` in
https://aiyprojects.withgoogle.com/vision/#makers-guide--tensorflow-model-compiler
